### PR TITLE
Embed license text in binary

### PR DESCRIPTION
**Pull request by kaidokert**
Rework license file API to provide it from embedded resource, rather than a bundled file.

b/74628213

### CODE REVIEW

File: cobalt/dom/navigator.cc [hunk 1]
```diff
@@ -153,49 +152,12 @@ Navigator::Navigator(script::EnvironmentSettings* settings,
...
-  // is smaller than 2^32.
-  int file_size = static_cast<int>(info.size);
-
-  std::unique_ptr<char[]> buffer(new char[file_size]);
-  SbFileReadAll(file, buffer.get(), file_size);
-  const std::string file_contents = std::string(buffer.get(), file_size);
-  SbFileClose(file);
-
-  return file_contents;
+  const char kEmbeddedLicens[] = "licenses_cobalt.txt";
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply cobalt/dom/navigator.cc]: Thanks for catching - was going too fast here.

File: cobalt/dom/navigator.cc [hunk 2]
```diff
@@ -153,49 +152,13 @@ Navigator::Navigator(script::EnvironmentSettings* settings,
...
-  // is smaller than 2^32.
-  int file_size = static_cast<int>(info.size);
-
-  std::unique_ptr<char[]> buffer(new char[file_size]);
-  SbFileReadAll(file, buffer.get(), file_size);
-  const std::string file_contents = std::string(buffer.get(), file_size);
-  SbFileClose(file);
-
-  return file_contents;
+  const char kEmbeddedLicenses[] = "licenses_cobalt.txt";
```
Comment (sherryzy) [inline cobalt/dom/navigator.cc:156]: super nit: maybe put this on the top (its original place)?

**Review by sherryzy — APPROVED** [review]
Looking good.

File: cobalt/dom/navigator.cc (hunk 2 above)
Comment (kaidokert) [reply cobalt/dom/navigator.cc]: Not a bad nit, provides a cleaner diff. Although i grabbed the code from `ParseUserAgentStyleSheet` which does what i did here.

**Pull request merged**